include/zephyr/drivers/accel2d.h

```diff
@@ -0,0 +1,680 @@
+	/** Draw any pixel within at least one path */
+	ACCEL2D_FILL_ANY,
+	/** Draw only pixels that reside within an odd number of paths */
+	ACCEL2D_FILL_ODD,
+};
+
+/**
+ * @brief 2D accelerator image rotation setting
+ *
+ * Note that the rotation setting here causes the 2D accelerator to treat the
+ * output buffer as rotated when copying pixels.
+ */
+enum accel2d_rotate {
+	/** Do not rotate output image */
+	ACCEL2D_ROTATE_NONE,
+	/** Rotate output image 90 degrees */
+	ACCEL2D_ROTATE_90,
+	/** Rotate output image 180 degrees */
+	ACCEL2D_ROTATE_180,
+	/** Rotate output image 270 degrees */
+	ACCEL2D_ROTATE_270,
+};
+
+/**
+ * @brief 2D accelerator scale factor
+ *
+ * Scales output buffer up or down by a fixed factor
+ */
+enum accel2d_scale {
+	ACCEL2D_SCALE_2,     /**< x2 */
+	ACCEL2D_SCALE_1_875, /**< x1.875 */
+	ACCEL2D_SCALE_1_75,  /**< x1.75 */
+	ACCEL2D_SCALE_1_625, /**< x1.625 */
+	ACCEL2D_SCALE_1_50,  /**< x1.50 */
+	ACCEL2D_SCALE_1_375, /**< x1.375 */
+	ACCEL2D_SCALE_1_25,  /**< x1.25 */
+	ACCEL2D_SCALE_1_125, /**< x1.125 */
+	ACCEL2D_SCALE_1,     /**< x1 */
+	ACCEL2D_SCALE_0_75,  /**< 0.75 */
+	ACCEL2D_SCALE_0_50,  /**< 0.5 */
+	ACCEL2D_SCALE_0_25,  /**< 0.25 */
+	ACCEL2D_SCALE_1_3,   /**< 1/3 */
+	ACCEL2D_SCALE_1_4,   /**< 1/4 */
+	ACCEL2D_SCALE_1_8,   /**< 1/8 */
+};
+
+/**
+ * @brief 2D accelerator capabilities flags
+ *
+ * Capabilities data for a 2D accelerator device.
+ */
+struct accel2d_capabilities {
+	/* Blending support */
+	unsigned int blend_support; /**< Bitmask of supported blending modes */
+	unsigned short alignment; /**< Alignment requirement for 2D accelerator buffers */
+	/* Basic API support */
+	unsigned int rect_support: 1; /**< support for accel2d_draw_rect */
+	unsigned int arc_support: 1;  /**< support for accel2d_draw_arc */
+	unsigned int line_support: 1; /**< support for accel2d_draw_line */
+	unsigned int blit_support: 1; /**< support for accel2d_blit_img */
+	/* Path API support */
+	unsigned int path_support: 1;   /**< support for accel2d path APIs */
+	unsigned int direct_support: 1; /**< support for accel2d_write_display */
+};
+
+/**
+ * @brief Draw a simple rectangle using the 2D accelerator
+ *
+ * Draws a simple rectangle using the 2D accelerator hardware.
+ * @param dev: 2D accelerator device object
+ * @param buf: 2D accelerator output buffer description
+ * @param src: Input source description.
+ * @param rect: rectangle output description
+ * @param blend: 2D accelerator porter-duff blend setting
+ * @retval 0 command was queued successfully
+ * @retval -ENOSYS if this 2D accelerator does not support this feature
+ * @retval -ENOTSUP if the 2D accelerator does not support the
+ *          requested combination of settings
+ * @retval -EINVAL invalid buffer alignment settings for the 2D accelerator
+ * @retval -EBUSY @ref accel2d_render must be called before this operation can
+ *          proceed, as no more rendering can be queued on the 2D accelerator
+ * hardware.
```

Comment:
Nit: Alignment, same for other API functions.

```suggestion
 * @retval -EBUSY @ref accel2d_render must be called before this operation can
 *          proceed, as no more rendering can be queued on the 2D accelerator
 *          hardware.
```